### PR TITLE
10 skohub shape

### LINKED ISSUE #10: SkoHub Shape
Build a shape that specifies SkoHub specific rules

### PR DESCRIPTION
**Pull request by sroertgen**
Adds a basic skohub shape, which is not dependent on SPARQL-based constraints and can therefore be used with tools like https://github.com/zazuko/rdf-validate-shacl

The shape is based on the types and the code used in SkoHub Vocabs and therefore a bit more restrictive than the regular SKOS shape.

Test Cases for each constraint were added and tests adjusted.

### CODE REVIEW

File: tests/invalid/skohub.shacl.ttl/11_c_no_language.ttl
Comment (acka47) [inline tests/invalid/skohub.shacl.ttl/11_c_no_language.ttl:1]: For easier understanding and to be consistent with other files, this file might better be named `11_c_prefLabel_no_language.ttl`.

**Comment by acka47** [comment]
I started to look into this. Looks very thorough on first glance. A general question: What does the numbering in the file names (e.g. `01_cs`, `26_c`) refer to? Is it documented somewhere?

**Comment by sroertgen** [comment]
good point. it is correlated to the order of the constraints in the shape. But I could add a comment to each constraint indicating the referenced test. What do you think?

sth like: 

```turtle
:ConceptSchemeShape
  a sh:NodeShape ;
  sh:targetClass skos:ConceptScheme ;
  sh:property [
    sh:path dct:title ;
    sh:minCount 1;
    sh:datatype rdf:langString ;
    sh:message "Concept Scheme has no dct:title with a language tag!" ;
    sh:severity sh:Violation ;
    rdfs:comment "This is SkoHub specific. Tested with: 01_cs_no_title.ttl" ;
  ] ;
```

File: skohub.shacl.ttl [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+:ConceptShape
+  a sh:NodeShape ;
+  sh:targetClass skos:Concept ;
+  sh:property [
+    sh:path skos:prefLabel ;
+    sh:minCount 1 ;
+    sh:datatype rdf:langString ;
+    sh:uniqueLang true ;
+    sh:message "Each skos:Concept has to provide a skos:prefLabel in a unique language" ;
+    sh:severity: sh:Warning ;
```
Comment (acka47) [inline skohub.shacl.ttl]: I think, this should also be a violation.

File: skohub.shacl.ttl [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+    sh:class skos:Concept ;
+    sh:path skos:narrowerTransitive ;
+    sh:severity sh:Violation ;
+    sh:message "Target class of skos:narrowerTransitive should be a skos:Concept" ;
+  ] ;
+  sh:property [
+    sh:path skos:narrowMatch ;
+    sh:nodeKind sh:IRI ;
+    sh:severity sh:Violation ;
+    sh:message "Target class of skos:narrowMatch should be a skos:Concept (IRI)" ;
```
Comment (acka47) [inline skohub.shacl.ttl]:
```suggestion
    sh:message "Target class of skos:narrowMatch should be a URI ;
```

File: skohub.shacl.ttl [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+    sh:path skos:broaderTransitive ;
+    sh:class skos:Concept ;
+    sh:severity sh:Violation ;
+    sh:message "Target class of skos:broaderTransitive should be a skos:Concept" ;
+  ] ;
+  sh:property [
+    sh:path skos:broadMatch ;
+    sh:nodeKind sh:IRI ;
+    sh:severity sh:Violation ;
+    sh:message "Target class of skos:broadMatch should be a skos:Concept" ;
```
Comment (acka47) [inline skohub.shacl.ttl]:
```suggestion
    sh:message "Target class of skos:broadMatch should be a URI" ;
```

File: skohub.shacl.ttl [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+    sh:path skos:related ;
+    sh:class skos:Concept ;
+    sh:severity sh:Violation ;
+    sh:message "Target class of skos:related should be a skos:Concept" ;
+  ] ;
+  sh:property [
+    sh:path skos:relatedMatch ;
+    sh:nodeKind sh:IRI ;
+    sh:severity sh:Violation ;
+    sh:message "Target class of skos:relatedMatch should be a skos:Concept (IRI)" ;
```
Comment (acka47) [inline skohub.shacl.ttl]:
```suggestion
    sh:message "Target class of skos:relatedMatch should be a URI" ;
```

File: skohub.shacl.ttl [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+  ] ;
+  sh:property [
+    sh:path skos:closeMatch ;
+    sh:nodeKind sh:IRI ;
+    sh:severity sh:Violation ;
+    sh:message "Target class of skos:closeMatch should be a skos:Concept (IRI)" ;
```
Comment (acka47) [inline skohub.shacl.ttl]:
```suggestion
    sh:message "Target class of skos:closeMatch should be a URI" ;
```

File: skohub.shacl.ttl [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+  ] ;
+  sh:property [
+    sh:path skos:exactMatch ;
+    sh:nodeKind sh:IRI ;
+    sh:severity sh:Violation ;
+    sh:message "Target class of skos:exactMatch should be a skos:Concept (IRI)" ;
```
Comment (acka47) [inline skohub.shacl.ttl]:
```suggestion
    sh:message "Target class of skos:exactMatch should be a URI" ;
```

File: skohub.shacl.ttl [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+    sh:path skos:inScheme ;
+    sh:class skos:ConceptScheme ;
+    sh:severity sh:Violation ;
+    sh:message "Target class of skos:inScheme should be a skos:ConceptScheme" ;
+  ] ;
+  sh:property [
+    sh:path skos:hasTopConcept ;
+    sh:class skos:ConceptScheme ;
+    sh:severity sh:Violation ;
+    sh:message "Target class of skos:hasTopConcept should be a skos:ConceptScheme" ;
```
Comment (acka47) [inline skohub.shacl.ttl]:
```suggestion
    sh:class skos:Concept ;
    sh:severity sh:Violation ;
    sh:message "Target class of skos:hasTopConcept should be a skos:Concept" ;
```

**Review by acka47 — CHANGES_REQUESTED** [review]
Requesting changes as commented/suggested.

File: skohub.shacl.ttl (hunk 7 above)
Comment (sroertgen) [reply skohub.shacl.ttl]: it should have been `skos:topConceptOf` anyway.

**Pull request merged**